n_t_with_s/Snakefile

```diff
@@ -0,0 +1,361 @@
+    chrm_list = [chr for chr in config["chrm_list"].split(",")]
+
+
+# This grabs the default mr from the first chromosome,
+# Ultimitely This needs to be replaced with the weighted average
+# of all chromosomes: This should be done in stdpopsim. 
+mutation_rate = species.genome.mean_mutation_rate
+
+
+# ###############################################################################
+# GENERAL RULES & GLOBALS
+# ###############################################################################
+
+
+seed_array = np.random.random_integers(1,2**31,replicates)
+genetic_map_downloaded_flag= ".genetic_map_downloaded"
+msmc_exec = "../ext/msmc/build/msmc"
+stairwayplot_code = "stairwayplot/swarmops.jar"
+
+try:
+    mask_file = config["mask_file"]
+except KeyError:
+    mask_file = None
+
+# Run full pipeline!
+#rule all:
+#   input: output_dir + "/Results/all_estimated_Ne.pdf"
+
+## Run only the simulations!
+#rule all:
+#    input: expand(output_dir + "/Intermediate/{seeds}/{chrms}.trees", seeds=seed_array, chrms=chrm_list) 
+
+## Run only stairwayplot and plot with coalescent rate!
+#rule all:
+#   input: output_dir + "/Results/stairwayplot.pdf"
+
+## Run only smc+++ 
+rule all:
+    input: expand(output_dir + "/Intermediate/{seeds}/{chrms}.trees.smc.gz", seeds=seed_array, chrms=chrm_list) 
+
+## Run only mcmc
+#rule all:
+#    input: output_dir+"/Results/msmc_estimated_Ne.png"
+
+rule download_genetic_map:
+    output: genetic_map_downloaded_flag
+    message: "Downloading default genetic map"
+    run:
+        # We need to have this here to avoid several threads trying to download the 
+        # the genetic map into the cache at the same time.
+        if genetic_map_id is not None:
+            genetic_map = species.get_genetic_map(genetic_map_id)
+            if not genetic_map.is_cached():
+                genetic_map.download()
+            with open(output[0], "w") as f:
+                print("File to indicate genetic map has been downloaded", file=f)
+
+
+rule simulation:
+    input: 
+        genetic_map_downloaded_flag,
+        # rules.top_directory.output,
+    output:
+        output_dir + "/Intermediate/{seeds}/{chrms}.trees"
+    # TODO get simulation parameters from stdpopsim, so we can share them easily 
+    # with anlysis code below?
+    threads: 1 
+    run:
+        contig = species.get_contig(wildcards.chrms, genetic_map=genetic_map_id)
+        samples = model.get_samples(*num_samples_per_population)
+        engine = stdpopsim.get_engine("slim") ## here we use the slim engine
+        ####### Here we are adding a given DFE
+        ####### Adding DFE/selection to the simulation  
+        dfe = species.get_dfe("Gamma_K17")
+        #contig.add_dfe(intervals=np.array([[0, int(contig.length)]]), DFE=dfe)
+
+        ## Adding annotation only seletion on exon region
+        exons = species.get_annotations("ensembl_havana_104_exons")
+        exon_intervals = exons.get_chromosome_annotations(wildcards.chrms).astype("int")
```

Comment:
doesn't this return `int` by default? i thought we fixed this?

Reply:
ops my bad, just checked and you are right: https://github.com/popsim-consortium/stdpopsim/blob/6e23ae5cf25934506e64f7e64110dce3b44c14ad/stdpopsim/annotations.py#L86

I was just following the [tutorial](https://github.com/popsim-consortium/stdpopsim/blob/main/docs/tutorial.rst), I will modify there accordingly too.